regxmllib/CMakeLists.txt

```diff
@@ -1,29 +1,92 @@
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.6)
 project (regxmllibc)
+option(BUILD_SHARED_LIB Bool OFF)
+
```

Comment:
cmake already defines [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/v3.5/variable/BUILD_SHARED_LIBS.html)

Reply:
Introduced for pure convenience when using a CMake GUI, from [1]:
This variable is often added to projects as an option() so that each user of a project can decide if they want to build the project using shared or static libraries

[1] https://cmake.org/cmake/help/v3.5/variable/BUILD_SHARED_LIBS.html

Reply:
Ok... but it says `BUILD_SHARED_LIB` and not `BUILD_SHARED_LIBS`.

Reply:
True, I forgot. It's a locally defined name.